oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/DocumentStoreIndexerBase.java

```diff
@@ -413,12 +413,13 @@ public void reindex() throws CommitFailedException, IOException {
                         }
                     }
                     log.info("Top slowest nodes to index (ms): {}", slowestTopKElements);
+                }
 
-                    indexerProviders.forEach(indexProvider -> {
-                        ExtractedTextCache extractedTextCache = indexProvider.getTextCache();
-                        CacheStats cacheStats = extractedTextCache == null ? null : extractedTextCache.getCacheStats();
-                        log.info("Text extraction cache statistics: {}", cacheStats == null ? "N/A" : cacheStats.cacheInfoAsString());
-                    });
+                for (NodeStateIndexerProvider indexerProvider : indexerProviders) {
+                    indexerProvider.close();
```

Comment:
Hm, shouldn't we close it after logging?
(I guess currently "indexerProvider.getTextCache()" will work after closing, but it seems wrong to me to close it first).

Reply:
Good point, it's a bad practise to access an object after closing it. I moved the close to the end of the block.